src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCombatAcademy.java

```diff
@@ -378,6 +378,30 @@ public BuildingEntry getBuildingRegistryEntry()
         return ModBuildings.combatAcademy;
     }
 
+    @Override
+    public void onWakeUp()
+    {
+        super.onWakeUp();
+        
+        final World world = getColony().getWorld();
+        if (world == null)
+        {
+            return;
+        }
+
+        for (final BlockPos pos : bedList)
+        {
+            BlockState state = world.getBlockState(pos);
+            state = state.getBlock().getExtendedState(state, world, pos);
+            if (state.getBlock() instanceof BedBlock
+                  && state.get(BedBlock.OCCUPIED)
+                  && state.get(BedBlock.PART).equals(BedPart.HEAD))
+            {
+                world.setBlockState(pos, state.with(BedBlock.OCCUPIED, false), 0x03);
+            }
+        }
+    }    
```

Comment:
Again this is supposed to be default behaviour for the entity.
Why was this not handled there.

Reply:
The entities do not handle this for any of our buildings.